### ISSUE TITLE
ERR_INVALID_PROTOCOL when using --skipSSLValidation

### ISSUE THREAD

**Issue by rostero1** [apollographql/apollo-tooling#928]
I have a self-signed cert and am trying to download the schema, but I get 

› apollo service:download --skipSSLValidation --endpoint=https://localhost/graphql/
  ✔ Loading Apollo Project
  ✖ Saving schema to schema.json

    → Protocol "https:" not supported. Expected "http:"
TypeError [ERR_INVALID_PROTOCOL]: Protocol "https:" not supported. Expected "http:"

**Comment by SpencerCarstens** [comment]
I am also getting this with the VSCode extension.

`TypeError [ERR_INVALID_PROTOCOL]: Protocol "https:" not supported. Expected "http:"
    at new ClientRequest (_http_client.js:109:11)
    at request (https.js:272:10)
    at /Users/---/.vscode-insiders/extensions/apollographql.vscode-apollo-1.4.0/node_modules/node-fetch/lib/index.js:1421:15
    at new Promise (<anonymous>)
    at fetch (/Users/---/.vscode-insiders/extensions/apollographql.vscode-apollo-1.4.0/node_modules/node-fetch/lib/index.js:1390:9)
    at /Users/---/.vscode-insiders/extensions/apollographql.vscode-apollo-1.4.0/node_modules/apollo-link-http/lib/bundle.umd.js:119:17
    at new Subscription (/Users/---/.vscode-insiders/extensions/apollographql.vscode-apollo-1.4.0/node_modules/zen-observable/lib/Observable.js:183:34)
    at Observable.subscribe (/Users/---/.vscode-insiders/extensions/apollographql.vscode-apollo-1.4.0/node_modules/zen-observable/lib/Observable.js:262:14)
    at /Users/---/.vscode-insiders/extensions/apollographql.vscode-apollo-1.4.0/node_modules/apollo-link/lib/bundle.umd.js:64:24
    at new Promise (<anonymous>)`

**Comment by sebastianseilund** [comment]
Same issue here. Instead of using the `--skipSSLValidation`/`-k` flag, you can set the `NODE_TLS_REJECT_UNAUTHORIZED` environment variable to `0`:

```sh
NODE_TLS_REJECT_UNAUTHORIZED=0 apollo service:download --endpoint <url>
```

Tested with Node.js v10.3.0. More info and kudos to [this stackoverflow answer](https://stackoverflow.com/questions/10888610/ignore-invalid-self-signed-ssl-certificate-in-node-js-with-https-request/21961005#21961005).

**Comment by Jayphen** [comment]
How would one set `NODE_TLS_REJECT_UNAUTHORIZED` for the VS Code extension? I am facing the same issue.

**Comment by marcosfede** [comment]
Any updates on this?

**Comment by trevor-scheer** [comment]
Closed via #1461

**Comment by woeterman94** [comment]
> How would one set `NODE_TLS_REJECT_UNAUTHORIZED` for the VS Code extension? I am facing the same issue.

set NODE_TLS_REJECT_UNAUTHORIZED=0
although it's not recommended.
